pie/storage/database.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+from typing import Dict, Optional, Union
+
+from sqlalchemy import BigInteger, Column, String
+
+from pie.database import database, session
+
+
+class Storage(database.base):
+    __tablename__ = "storage_data"
+
+    module = Column(String, primary_key=True)
+    guild_id = Column(BigInteger, primary_key=True)
+    key = Column(String, primary_key=True)
+    value = Column(String)
+
+    @staticmethod
+    def set(
+        module: str,
+        key: str,
+        value: str,
+        guild_id: int = 0,
+        ignore_existing: bool = False,
+    ) -> Optional[Storage]:
+        data = (
+            session.query(Storage)
+            .filter_by(module=module)
+            .filter_by(key=key)
+            .filter_by(guild_id=guild_id)
+            .one_or_none()
+        )
+
+        if data and ignore_existing:
+            return None
+
+        if not data:
+            data = Storage(module=module, key=key, guild_id=guild_id)
+
+        data.value = value
+        session.merge(data)
+        session.commit()
+
+        return data
+
+    @staticmethod
+    def get(module: str, key: str, guild_id: int = 0) -> Optional[str]:
```

Comment:
Why is it default to `0`? If it's for DMs, I'd let modules handle it themslves, we don't have many default values in table methods.

Reply:
It's for global settings.

Reply:
I'd leave it to the developer to decide it. I'm not fully comfortable with global setting being the default.

Reply:
OK, done.